infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -516,7 +520,9 @@ private synchronized void handleResetLogUnit(RequestMsg req, ChannelHandlerConte
 
         serverContext.setLogUnitEpochWaterMark(req.getPayload().getResetLogUnitRequest().getEpoch());
 
-        batchWriter.addTask(BatchWriterOperation.Type.RESET, req)
+        batchProcessor.restart();
+
+        batchProcessor.addTask(BatchWriterOperation.Type.RESET, req)
```

Comment:
> But right now we are not trying to solve your more complex issue, we are trying at least to solve the issue that happens in production.

That's the point I'm trying to make, I don't think this patch solves much because it targets such a specific case that doesn't happen in isolation. Essentially, it's replacing one problem with another. Instead of having a node that can't accept writes to be part of the active chain, you have a node that keeps leaving/rejoining the chain continuously.